ansys_optical_automation/interop_process/Coating_converter_speos_zemax_Analysis.py

```diff
@@ -0,0 +1,398 @@
+            for wavelength_index in range(nb_wavelength):
+                offset = angle_index + wavelength_index * (nb_angle_of_incidence + 1)
+                reflectance_ppol_1 = float(coating_data[offset][2])
+                transmittance_ppol_1 = float(coating_data[offset][4])
+                file_id1.write(myformat.format(100 * reflectance_ppol_1) + '\t' + myformat.format(
+                    100 * transmittance_ppol_1) + '\t')
+            file_id1.write("\n\t")
+            for wavelength_index in range(nb_wavelength):
+                offset = angle_index + wavelength_index * (nb_angle_of_incidence + 1)
+                reflectance_spol_1 = float(coating_data[offset][1])
+                transmittance_spol_1 = float(coating_data[offset][3])
+                file_id1.write(myformat.format(100 * reflectance_spol_1) + "\t" + myformat.format(
+                    100 * transmittance_spol_1) + "\t")
+            file_id1.write("\n")
+        file_id1.close()
+        # print("File " + coatingfilename1 + " created")
+        coating_data.clear()
+        # os.remove(resultfullfilename)
+
+
+    def __init__(self,coatingfilename, coatingfolder,
+            substrate_catalog,substrate_name,
+            user_wavelength_min, user_wavelength_max, nb_wavelength, speos_wavelength_units_um,
+            nb_digits, skip_lines):
+        """
+            function that converts a Zemax coating / substrate into a speos coating file and a bsdf180
+            The function reads the data from the Transmission vs Angle analysis
+            The settings are not implemented in ZOS-API for this analysis.
+            So the settings by default should be set to
+            - incident angles  from 0 to 90
+            - surface to 1
+
+            Parameters
+            ----------
+            coatingfilename : string
+                Name of the coating file to convert
+            coatingfolder : string
+                Path to the coating file
+            substrate_catalog : string
+                Catalog of the materials
+            substrate_name : string
+                Names of the materials
+            user_wavelength_min : float
+                Minimum user wavelength
+            user_wavelength_max : float
+                Maximum user wavelength
+            nb_wavelength = integer
+                Number of wavelengths
+            speos_wavelength_units_um : integer
+                Conversion value to go from speos units to um. For example if speos is in nm, it is 1000
+            nb_digits : integer
+                Number of digits of the converted data
+            skip_lines: integer
+                number of lines that are ignored at each loop - angle of incidence resolution
+            Returns
+            -------
+
+            """
+
+        zos = BaseZOS()
+        # load local variables
+        zosapi = zos.zosapi
+        the_application = zos.the_application
+        the_system = zos.the_system
```

Comment:
this variable is also used in check_wavelength_range, thus this can be an internal variable

Reply:
Do you mean the_system?
I am defining this variable inside the __init__ function and then use it as an argument in the check_wavelength_range function. What would you like me to do?